File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        int prev1 = 1;
        int prev2 = 0;
        int current = 0;
        if(n <2) return n;
        for(int i=2;i< n+1; i++ ){
            current = prev1 + prev2;
            prev2 = prev1;
            prev1 = current;
        }        
        return prev1;
    }
}

File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        #Recursion      return n if n<2 else self.fib(n-1) + self.fib(n-2)
        #Memoization
        if n < 2: return n 
        prev1, prev2 = 1, 1
        for i in range(2,n+1):
            prev1, prev2 = prev1 + prev2, prev1 
        return prev2
        
        
            
        
            
        